tensorflow/python/distribute/input_lib_type_spec_test.py

```diff
@@ -450,6 +450,89 @@ def f(v):
       f(v)
     self.assertEqual(self.trace_count, 1)
 
+  @combinations.generate(
```

Comment:
Does this test fail without your fixes in input_lib.py?

Reply:
Yes it fails:
ValueError: The two structures don't have the same nested structure.